src/java/main/org/apache/zookeeper/client/ZooKeeperSaslClient.java

```diff
@@ -590,14 +590,13 @@ public boolean clientTunneledAuthenticationInProgress() {
                 // authentication is either in progress, successful, or failed.
 
                 // 1. Authentication hasn't finished yet: we must wait for it to do so.
-                if ((isComplete() == false) &&
-                    (isFailed() == false)) {
+                if ((!isComplete()) && (!isFailed())) {
                     return true;
                 }
 
                 // 2. SASL authentication has succeeded or failed..
                 if (isComplete() || isFailed()) {
```

Comment:
Maybe i understand what you mean,  if **`not_complete && not_failed`** (Line: 593) is `FALSE` means **`complete || non_failed`** and  **`complete || failed`** and **`non_complete || failed`** is `TRUE`, then the `isComplete() || isFailed()` (Line: 599) will always be `TRUE`.